<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RegisterWindow</class>
 <widget class="QDialog" name="RegisterWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>用户注册</string>
  </property>

  <!-- ✅允许用户拖拽右下角改变大小 -->
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>

  <!-- ✅确保窗口可以自由调整大小 -->
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>500</height>
   </size>
  </property>

  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1,2,0,0">
   <item>
    <widget class="QLabel" name="label_title">
     <property name="text">
      <string>新用户注册</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="styleSheet">
      <string notr="true">font-size: 20px; font-weight: bold; padding: 10px;</string>
     </property>
    </widget>
   </item>

   <item>
    <widget class="QGroupBox" name="groupBox_user">
     <property name="title">
      <string>用户信息</string>
     </property>

     <!-- ✅允许内容扩展 -->
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>

     <layout class="QFormLayout" name="formLayout_user">
      <item row="0" column="0">
       <widget class="QLabel" name="label_username">
        <property name="text">
         <string>用户名*</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="edit_username">
        <property name="placeholderText">
         <string>3-50个字符</string>
        </property>
       </widget>
      </item>

      <item row="1" column="0">
       <widget class="QLabel" name="label_password">
        <property name="text">
         <string>密码*</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="edit_password">
        <property name="placeholderText">
         <string>至少6个字符</string>
        </property>
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>

      <item row="2" column="0">
       <widget class="QLabel" name="label_confirm_password">
        <property name="text">
         <string>确认密码*</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="edit_confirm_password">
        <property name="placeholderText">
         <string>再次输入密码</string>
        </property>
        <property name="echoMode">
         <enum>QLineEdit::Password</enum>
        </property>
       </widget>
      </item>

      <item row="3" column="0">
       <widget class="QLabel" name="label_phone">
        <property name="text">
         <string>手机号*</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="edit_phone">
        <property name="placeholderText">
         <string>11位手机号码</string>
        </property>
       </widget>
      </item>

      <item row="4" column="0">
       <widget class="QLabel" name="label_email">
        <property name="text">
         <string>邮箱</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="edit_email">
        <property name="placeholderText">
         <string>可选</string>
        </property>
       </widget>
      </item>

      <item row="5" column="0">
       <widget class="QLabel" name="label_real_name">
        <property name="text">
         <string>真实姓名</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="edit_real_name">
        <property name="placeholderText">
         <string>可选</string>
        </property>
       </widget>
      </item>

     </layout>
    </widget>
   </item>

   <item>
    <widget class="QGroupBox" name="groupBox_vehicle">
     <property name="title">
      <string>车辆信息（可添加多辆）</string>
     </property>

     <!-- ✅可扩展 -->
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>2</verstretch>
      </sizepolicy>
     </property>

     <layout class="QVBoxLayout" name="vehicleMainLayout" stretch="1,0">
      <item>
       <widget class="QScrollArea" name="scrollArea_vehicles">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="vehiclesContainer">
         <layout class="QVBoxLayout" name="vehicleListLayout"/>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_add_vehicle">
        <property name="text">
         <string>添加车辆</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_buttons">
     <item>
      <widget class="QPushButton" name="btn_register">
       <property name="text">
        <string>注册</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_back">
       <property name="text">
        <string>返回登录</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>

   <item>
    <widget class="QLabel" name="label_status">
     <property name="text">
      <string></string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="styleSheet">
      <string notr="true">color: red; padding: 5px; font-size: 12px;</string>
     </property>
    </widget>
   </item>

  </layout>
 </widget>

 <resources/>
 <connections/>
</ui>
